pkg/apps/application.jsx

```diff
@@ -27,20 +27,36 @@ import { Page, PageBreadcrumb, PageSection } from "@patternfly/react-core/dist/e
 import { Stack } from "@patternfly/react-core/dist/esm/layouts/Stack/index.js";
 import { ExternalLinkAltIcon } from '@patternfly/react-icons';
 
-import * as PackageKit from "./packagekit.js";
-import { icon_url, launch, ProgressBar, CancelButton } from "./utils";
+import { icon_url, launch, ProgressBar, CancelButton, reload_bridge_packages, ProgressReporter } from "./utils";
 
 import "./application.scss";
+import { getPackageManager } from "packagemanager.js";
 
 const _ = cockpit.gettext;
 
+async function install_package(pkgname, progress_cb) {
+    const packagemanager = await getPackageManager();
+    await packagemanager.install_packages([pkgname], progress_cb);
+    await reload_bridge_packages();
+}
+
+async function remove_package(filename, progress_cb) {
+    const progress = new ProgressReporter(0, 1, progress_cb);
+    const packagemanager = await getPackageManager();
+    const pkgnames = await packagemanager.find_file_packages([filename], progress.progress_reporter);
+    progress.base = 1;
+    progress.range = 99;
+    await packagemanager.remove_packages(pkgnames, progress.progress_reporter);
+    await reload_bridge_packages();
+}
+
 export const ActionButton = ({ comp, progress, action }) => {
     function install(comp) {
-        action(PackageKit.install, comp.pkgname, _("Installing"), comp.id);
+        action(install_package, comp.pkgname, _("Installing"), comp.id);
     }
 
     function remove(comp) {
-        action(PackageKit.remove, comp.file, _("Removing"), comp.id);
+        action(remove_package, comp.file, _("Removing"), comp.id);
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-22611-83e71f7c-20251121-122204-fedora-43-devel/Coverage/lcov/github-pr.diff.gcov.html)